### PR TITLE
FF100 Add AbortSignal.timeout() static method

### PR DESCRIPTION
**Pull request by hamishwillee**
FF100 adds support for `AbortSignal.timeout()` in https://bugzilla.mozilla.org/show_bug.cgi?id=1753309

This adds docs for timeout, and also updates the `AbortSignal` parent with more examples showing how it is used. Also important, showing the limits.

Specifically the `timeout` is  static method that returns a self aborting `AbortSignal` that aborts with `TimeoutError` after a specific time. Note that this can also be aborted with `AbortError` by pressing a browser stop button (if it has one) or presumably by shutting down the associated tab if this was called in a worker or whatever. 

However this has no associated `AbortController` so you can't call `abort()` to trigger this signal. Since `fetch()` takes just one signal, this means you can't have your own button for aborting and a timeout using this (ety).

I have tried to capture all that, and added a release note.

### CODE REVIEW

File: files/en-us/web/api/abortsignal/timeout/index.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - : The "active" time in milliseconds before the returned {{domxref("AbortSignal")}} will abort.
+
+### Return value
+
+An {{domxref("AbortSignal")}}.
+The signal will abort with its {{domxref("AbortSignal.reason")}} property set to a `TimeoutError` {{domxref("DOMException")}}.
+
+## Examples
+
+A simple example showing a fetch operation that will timeout if unsuccessful after 5 seconds, is shown below.
```
Comment (hamishwillee) [inline files/en-us/web/api/abortsignal/timeout/index.md:43]: Note, this just shows how THIS API is used. I don't show how to do abort+timeout case, that is covered in AbortSignal.

File: files/en-us/web/api/abortcontroller/index.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ You can create a new `AbortController` object using the {{domxref("AbortControll
 
 ## Examples
 
+> **Note:** There are additional examples in the {{domxref("AbortSignal")}} reference.
```
Comment (hamishwillee) [inline files/en-us/web/api/abortcontroller/index.md:35]: The example below is a duplicate of the original example in `AbortSignal`. When I added the new examples in `AbortSignal`, I decided not to duplicate everything and added this note to point to the new examples. 

But, should I remove this whole section and just have one set of docs either hear or in AbortSignal? I am tempted.

Comment (Rumyra) [reply files/en-us/web/api/abortcontroller/index.md:35]: I think as long as the examples are relevant to the pages they pertain to this is ok - there's plenty of dupe examples through out the docs 🤷🏻‍♀️

**Review by Rumyra — APPROVED** [review]

**Comment by Rumyra** [comment]
All looks good to mw @hamishwillee 👍

**Pull request merged**